### PR TITLE
Send success response when there are no sessions. Expose logout error…

### PR DESCRIPTION
**Pull request by mcastany**
… handler

Changes in this PR:
- Expose logoutError functionality to response in a more compliant way when there's an error in the logout flow
- When there is no session but there's information in the options object to reply to return success

### CODE REVIEW

File: lib/logout.js [hunk 1]
```diff
@@ -71,7 +71,8 @@ module.exports.logout = function (options) {
           inResponseTo:   data.id,
           destination:    data.serviceProviderLogoutURL || options.destination, // Destination taken from session (matches issuer from the LogoutRequest with session serviceProviderId)
           issuer:         options.issuer, // IdP is the Issuer for this LogoutResponse
-          samlStatusCode: isPartialLogout ? STATUS.PARTIAL_LOGOUT : STATUS.SUCCESS
+          samlStatusCode: isPartialLogout ? STATUS.PARTIAL_LOGOUT : STATUS.SUCCESS,
+          samlStatusMessage: options.samlStatusMessage
```
Comment (mcastany) [inline lib/logout.js]: Remove. Not needed in logout flow. Only for error messages

File: lib/index.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 exports.auth            = require('./samlp').auth;
 exports.logout          = require('./logout').logout;
+exports.logoutError     = require('./logout').logoutError;
```
Comment (ziluvatar) [inline lib/index.js]: Modify it to `sendLogoutError`

File: lib/logout.js [hunk 3]
```diff
@@ -350,4 +351,30 @@ function prepareAndSendToken(req, res, element_type, token, options, cb) {
...
+      inResponseTo:   options.id,
+      destination:    options.destination,
+      issuer:         options.issuer,
+      samlStatusCode:  error.code || STATUS.RESPONDER,
+      samlStatusMessage: error.description
+    });
+
+    // Signature location
+    options.reference = constants.ELEMENTS.LOGOUT_RESPONSE.SIGNATURE_LOCATION_PATH;
+    options.relayState = options.relayState || req.query.RelayState || (req.body || {}).RelayState || '';
```
Comment (ziluvatar) [inline lib/logout.js]: `|| ''` not needed, `prepareAndSendToken` does it https://github.com/auth0/node-samlp/blob/df1982e/lib/logout.js#L318

File: lib/logout.js [hunk 4]
```diff
@@ -350,4 +351,30 @@ function prepareAndSendToken(req, res, element_type, token, options, cb) {
...
+    if (!options.destination) {
+      return next(new Error('Destination not specified'));
+    }
+
+    var error = options.error || {};
+    
+    var logoutResponse = templates.logoutresponse({
+      id:             '_' + utils.generateUniqueID(),
+      instant:        utils.generateInstant(),
+      inResponseTo:   options.id,
```
Comment (ziluvatar) [inline lib/logout.js]: What is this `id` coming from?

File: lib/logout.js (hunk 1 above)
Comment (mcastany) [inline lib/logout.js]: Removing this variable fails when compiling the template. That's why I added it. If empty message won't be set

File: lib/logout.js (hunk 4 above)
Comment (mcastany) [inline lib/logout.js]: Renamed to `options.inResponseTo`

**Pull request merged**